Use ring buffer in DisplayTask

Ichbinjoe opened the pull request:
Also modifies DisplayTask to use Java thread interrupts for control
throw, which integrates with held locks and sleeps to properly die
immediately.

Partially reverts #7

Ichbinjoe committed:
Use ring buffer in DisplayTask

Also modifies DisplayTask to use Java thread interrupts for control
throw, which integrates with held locks and sleeps to properly die
immediately.

pop4959 commented on src/main/java/me/mattstudios/holovid/display/BufferedDisplayTask.java:
```diff
@@ -1,19 +1,13 @@
-import net.minecraft.server.v1_16_R1.ChatHexColor;
-import net.minecraft.server.v1_16_R1.ChatModifier;
-import net.minecraft.server.v1_16_R1.IChatBaseComponent;
+import net.minecraft.server.v1_16_R1.*;
```
yes import entire nms

Ichbinjoe replied:
frickin intellij

Ichbinjoe committed:
Remove cleanup, fix edgecases with stop

Stop had an edgecase where stop() could be called before run() had been
which would have 1) caused a NPE & 2) would not have stopped the task.
Add a lock as well as a DOA flag.

Ichbinjoe committed:
Un-intellij some imports

Chasewhip8 commented:
"Partialy" the most important fix is there lol

Ichbinjoe committed:
Explicitly initialize lastDisplayed

This is a hack to make the time delay logic work. This was not explicit
before, but this makes the hack explicit.

The only time this wouldn't work would be when the server's clock is set
to Jan  1st 1970, in which case you would wrongly wait one frame delay.

Horrible, isn't it.

Ichbinjoe commented:
Thread interruption is a good tool that can be used for weird stuff like this.
